frontend/src/components/Picasso.vue

```diff
@@ -3,28 +3,30 @@ Listing all given event instances in a calendar view.
 -->
 
 <template>
-  <v-calendar class="ec-picasso"
-              :events="eventInstances"
-              event-start="start_time"
-              event-end="end_time"
-              :event-name="getEventName"
-              :event-color="getEventColor"
-              interval-height="42"
-              :interval-format="getIntervalFormat"
-              first-interval="5"
-              interval-count="19"
-              :start="startDateString"
-              :end="endDateString"
-              locale="de-ch"
-              :day-format="dayFormat"
-              :type="type"
-              :weekday-format="weekdayFormat"
-              :weekdays="[1, 2, 3, 4, 5, 6, 0]"
-              color="primary"
-              @click:event="showEventInstance" />
+  <v-calendar
+    class="ec-picasso"
+    :events="eventInstances"
+    event-start="start_time"
+    event-end="end_time"
+    :event-name="getEventName"
+    :event-color="getEventColor"
```

Comment:
Theoretisch kann man in Vue auch hier in Props den `loading` Filter benutzen. Hier ist das leider nicht ganz so einfach, weil wir eine Funktion in die Prop reingeben. Würden wir den bestehenden Filter benutzen, so bekäme er nur die Funktion `getEventName` als Argument, nicht das Resultat der Funktion. Man könnte aber den Filter so erweitern, dass er merkt wenn er eine Funktion als Input bekommt und diese wrappt. Ich habe das mal so versucht:
```js
Vue.filter('loading', function (value, loadingState, isLoading = v => typeof v === 'function' && v.loading) {
  if (typeof value === 'function' && !value.loading) {
    // Wrap the function that is passed into the | loading filter
    return (v, ...args) => isLoading(v) ? loadingState : value(v, ...args)
  }
  return isLoading(value) ? loadingState : value
})
```

Dann könnten wir die loading Logik aus `getEventName` und `getEventColor` rausnehmen. Die passt in meinen Augen nicht wirklich dort hin.

```suggestion
    :event-name="getEventName | loading('Lädt…', ({ input }) => input.event()._meta.loading)"
    :event-color="getEventColor | loading('grey lighten-2', ei => ei.event()._meta.loading)"
```

Reply:
Ja wenn das so gut funktioniert, dann kann ich das machen. Aber @carlobeltrame ich verstehe den Code noch nicht.

Reply:
Ja ist leider nicht mehr ganz simpel. Ich probiere es zu erklären.
* Ich würde am liebsten schreiben: `:event-name="getEventName | loading('Lädt...')"`.
* Vuetify erwartet in der `event-name` Prop eine Funktion, also muss `getEventName | loading('Lädt...')` eine Funktion als Resultat haben.
* Wenn man `getEventName | loading('Lädt...')` schreibt, dann ruft Vue unseren Filter mit der Funktion `getEventName` als erstes Argument (`value`) und dem String `'Lädt...'` als zweites Argument (`loadingState`) auf.
* Jetzt müssen wir als erstes mal erkennen, ob da jemand einen normalen Wert (`camp.name | loading(...)`) oder eine Funktion (`getEventName | loading(...)`) in den `loading` Filter füttert. Wenn der Typ des ersten Arguments `function` ist, dann könnte es immer noch sein, dass das nur ein vorläufiges Resultat eines loadingProxies ist. Darum schauen wir zusätzlich noch, ob das `loading` Flag abwesend ist, welches vom loadingProxy gesetzt werden würde.
* Im `if` drin wissen wir also, dass eine echte, bekannte Funktion in den `loading` Filter gegeben wurde. Also müssen wir jetzt auch eine Funktion als Resultat ausspucken.
* Unsere resultierende Funktion muss ein oder mehrere Argumente akzeptieren, genau so viele wie `getEventName` nehmen würde. Wir wissen aber nicht genau wie viele das sind, daher verwenden wir `...args` um einfach alle verfügbaren Argumente in einem Array zu bekommen.
* Uns interessiert später das erste Argument besonders, darum benenne ich es separat `v`.
* Nun muss ich in der neuen Funktion drin entscheiden, ob die Argumente (z.B. die EventInstance die Vuetify da reinfüttern will) noch am laden sind oder nicht. Das Problem ist: Die Definition von "geladen" kann von Benutzung zu Benutzung des Filters unterschiedlich sein. Bei `getEventName` gilt der Input als geladen, sobald `v.input.event()` geladen ist. Bei `getEventColor` hingegen sobald `v.event()` geladen ist. Da ich das nicht im `loading` Filter drin wissen kann, biete ich die Möglichkeit, von aussen (via `isLoading`) dieses Verhalten zu steuern.
* Falls `isLoading` befindet, dass das erste Argument (`v`) noch am laden ist, dann gebe ich den `loadingState` als Resultat an Vuetify zurück.
* Falls `isLoading` hingegen befindet, dass das erste Argument fertig geladen ist, kann ich es durch die eigentliche Funktion (`getEventName`, welche ja in `value` steht) füttern und das tatsächliche finale Resultat an Vuetify zurückgeben.

Weiss jetzt nicht ob dir die Textwand viel hilft. Vielleicht fällt mir auch noch ein Weg ein, den Code etwas aufzusplitten und so verständlicher zu machen.